packages/vue/vue-vanilla/src/util/composition.ts

```diff
@@ -28,7 +28,15 @@ export const useVanillaControl = <
 
   const isFocused = ref(false);
   const onChange = (event: Event) => {
-    input.handleChange(input.control.value.path, adaptTarget(event.target));
+    let data = adaptTarget(event.target);
+
+    if (appliedOptions.value.removeWhenEmpty) {
+      if (isEmpty(data, input.control.value.schema.type)) {
+        data = undefined;
+      }
+    }
```

Comment:
Also there is the question whether we want to be able to configure each type of property separately. The typical things people want to remove are:
- `string` when they are equal to `''`
- `object` when it only has `undefined` properties
- `array` when it is empty

I'm not sure about other types like `number` and `boolean` and whether we should also handle setting to `null` instead of `undefined`. And when `null` whether we should check the JSON Schema whether that is even a valid value for the given type.

So the configuration could for example look like this: `removeWhenEmpty: { string: true, object: true, array: false}` or even like this `transformWhenEmpty: { string: 'undefined', object: null, array: false}`.

This configuration could be global and then also be overwritten locally with `uischema.options`.